projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt

```diff
@@ -116,6 +116,8 @@ class ProjectorServer private constructor(
 
   private val markdownQueue = ConcurrentLinkedQueue<ServerMarkdownEvent>()
 
+  private val baseQueue = ConcurrentLinkedQueue<ServerEvent>()
```

Comment:
I'm not sure, there are too many queues now, but this one is just generic. Maybe it's better to make it specific as other queues:

```suggestion
  private val browseUriQueue = ConcurrentLinkedQueue<ServerBrowseUriEvent>()
```

Reply:
I don't see a big difference between two. Maybe all queues should be just merged in one

Reply:
I don't mind merging the queues. However `baseQueue` still sounds unclear. Maybe we should name that merged queue like `highLevelEvents` (there are caret, markdown, and uri events) or just `commonQueue` (this queue contains the events that should go to all the connected users)

Reply:
I like `commonQueue`. Will update the PR at the end of the week